neo/Consensus/ConsensusContext.cs

```diff
@@ -149,5 +149,17 @@ public void Reset(Wallet wallet)
             }
             _header = null;
         }
+
+        public bool VerifyRequest()
+        {
+            if (!State.HasFlag(ConsensusState.RequestReceived))
+                return false;
+            if (!Blockchain.GetConsensusAddress(Snapshot.GetValidators(Transactions.Values).ToArray()).Equals(NextConsensus))
+                return false;
+            Transaction tx_gen = Transactions.Values.FirstOrDefault(p => p.Type == TransactionType.MinerTransaction);
+            Fixed8 amount_netfee = Block.CalculateNetFee(Transactions.Values);
+            if (tx_gen?.Outputs.Sum(p => p.Value) != amount_netfee) return false;
+            return true;
+        }
```

Comment:
Seems correct to me. The `NextConsensus` part is the same, Request should have been received too (otherwise there's no Miner tx). Outputs of Miner tx must be exactly network fee, seems nice.